arduino-ide-extension/src/node/monitor-manager.ts

```diff
@@ -11,16 +11,34 @@ import {
 
 type MonitorID = string;
 
+type UploadState = 'uploadInProgress' | 'pausedForUpload' | 'disposedForUpload';
+type MonitorIDsByUploadState = Record<UploadState, MonitorID[]>;
```

Comment:
open to ideas on better naming, essentially a state object, it groups monitor ids based on their "UploadState"